---
title: "Cancer transcriptome cleaning by deconvolution"
author: "Jonas Freimuth"
output:
  html_document:
    toc: true
    toc_depth: 2
bibliography: citations/cancer-cleaning_File.bib
---

# Introduction

Due to its relatively low cost, bulk RNA sequencing (bulk RNA-seq) has seen
widespread adoption in clinical cancer studies [@hong.2020]. While there
currently is a boom in single-cell as well as spatial transcriptomics, which
allow for very detailed expression profiling on the level of individual
cells, the associated cost remains a factor preventing these technologies from
being a resource of large-scale, high quality data-sets. Such resources would,
however, be of great use in generating new insights in cancer research when
combined with the wealth of machine-learning applications developed over the
last decades (e.g. @garg.2021, @newell.2022). With the limitations of the more
advanced methods in mind, efforts to improve the usability of tumor bulk RNA-seq
data for machine-learning tasks would be well spent.

However, the resolution of bulk RNA-seq data compared to the other methods
mentioned above is low. For tumor samples this means that bulk RNA-seq profiles
do not represent pure expression of cancerous cells, as a tumor tissue sample
usually also includes other cell types. These may be cells belonging to the
surrounding tissue (possibly due to imprecision during biopsy), or immune cells
(like tumor infiltrating lymphocytes). These cells constitute the tumor
microenvironment (TME), whose influence on disease progression [@barkley.2022]
and as a treatment target itself [@xiao.2021] is still being studied. When
sequencing tumor tissue samples, the transcripts expressed in these non-tumor
cells become incorporated into the expression data [@li.2021], which can
represent confounding for downstream predictive modeling taking. If the
expression profiles of the TME and of the cancer cells could be separated, the
performance of such models may be increased.

Ever since the widespread adoption of bulk gene expression quantification
technologies in the form of microarrays, researchers had to deal with the
problem of dissecting the combined expression of a mixtures of cells
[@ghosh.2003, @stuart.2004]. Methods developed for microarray deconvolution





 approaches to estimate the expression profiles of
individual cell types from RNA-seq data. Most older methods addressing the issue
of disentangling tumor expression from other cells, like DeMix [@ahn.2013],
ISOpure [@aran.2017], are designed for expression profiling arrays and may not
generalize to the unlimited number of genes made accessible by bulk RNA-seq.
Also, they were designed to only distinguish between two or few cell types /
conditions and may not be suited to dissect the cancer cells from the specific
mix of cell types surrounding them. 






and I am aware of two methods using bulk RNA-seq data, whose primary objective
is to estimate cell type proportion. These are Cibersortx [@newman.2019], and
BayesPrism [@chu.2022].
[Cibersortx ist doof und erfordert mehr bulks wie cell types, BayesPrism computationally intensive]

Cell type deconvolution already exists as a way to estimate the proportions of
cell types contributing to the expression profile of a bulk tissue RNA-seq
sample. One of the basic approaches consists of solving a system of linear
equations, minimizing the distance between the transcript predictions and the
bulk data. Under the assumption that cancer cells vary stronger in their
expression than the other cells in the microenvironment, we can expect the
residuals to contain much of this excess variation, thus representing a way to
“clean” the bulk cancer expression of the expression of non-cancer cells. This
method requires average expression profiles.

[High variability of cancer cells vs. remaining TME cells.]
[Compare to BayesPrism only as it works natively in R while BLADE is python, Cibersortx is container]

* Average expression profiles derived from scRNA-seq data
* Issue of increasing distance between actual cell expression and average
  expression profile.
* A way to possibly “boost” the signal of pure cancer expression, would be to
  not include the cancer cell type in the reference. This would of course 
  decrease the accuracy of the deconvolution model for predicting cell type
  proportions, but would in theory increase the variance in expression due to
  cancer cells that is not explained by the model, and which would therefore end
  up in the residuals. [I think however that it's not straightforward which
  portions end up in the residuals. There is still a fraction of each expression
  which gets “mis-attributed” to another cell type, but which counts as
  accounted for and thus is not present in the residuals.]

Research questions:

* Can the residuals from basic linear deconvolution methods be used to infer
  information about the tumor and inform treatment?
* Are the residuals gathered this way superior to using plain bulk data?

# Methods & Data

## Data

Single cell RNA sequencing data consists of transcript count data breast cancer
tumor samples from @wu.2021 (Gene Expression Omnibus, GEO accession
[GSE176078](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE176078)). It
contains 100'064 cells, 29'733 genes, from 26 samples. The cells are annotated
as belonging to one of nine cell types, with further sub-type information
available. A subset of 20 tumor samples has been processed via bulk RNA-seq.

For prediction tasks, bulk RNA-seq data of breast cancers annotated with
clinical data was obtained from the Cancer Genome Atalas (TCGA). The data was
comprised of all 1'095 patient samples from the TCGA-BRCA project with bulk
RNA-seq data available [Ensure reproducibility]. For the survival prediction
task, survival endpoint data derived from curated clincial data by @liu.2018 was
retrieved for the corresponding tumor samples.

## Methods - Transcriptome cleaning overview

To clean a bulk RNA-seq transcriptome, you first need to run a deconvolution on
it.

[Explanation of deconvolution as least squares problem & why nnls is sensible.]
[Mention other, more advanced methods.]
[Mathematical explanation.]
[Problem using cleaned bulk RNA-seq profiles as reference: It would need to be ensured that each profile represents a single-cell.]

For this you need a Reference and the bulk RNA-seq data to deconvolute.

### Reference

Here, the Reference is always made up averaged expression profiles derived from
scRNA data. For this, the per gene expression counts from @wu.2021 were averaged
across all cells of the same cell type. Since the dataset provided by @wu.2021
is already filtered for high quality cells, no cell quality control was
performed by me.

To explore the effects of restricting the set of genes used for deconvolution on
estimation accuracy for both cell type abundances and cancer expression, a set
of selection criteria was applied to the genes of the Reference. (First was
basic gene quality control, which comprised filtering the genes to be expressed
in at least 5 cells, removal of )

Second, each gene was assigned a marker score, quantifying its likely usefulness
as a gene in the deconvolution reference. Genes with a good score would have an
expression signature across cell types which would allow for clear
identification of just a single cell type by having particularly high or low
expression in just that cell type. Multiple ways to determine this score were
explored during preliminary experiments (DESeq2 @love.2014, Hampel filter
@davies.1993), but ultimately a simple Wilcoxon test [wilcoxon.1945] for
comparing each cell type to the pooled remaining cell types (via
`FindAllMarkers` from the R Seurat package v.5, @hao.2023) was chosen for its
combination of relative simplicity and accuracy. The marker score for the
Wilcoxon test based marker selection was the test p-Value (adjusted for multiple
testing via the Holm-Bonferroni method [@holm.1979]). Only genes differentially
expressed in a single cell type with at $p_{adj} < 0.05$ were considered as
marker genes, with lower values being considered better. It was contrasted with
choosing a random value for the marker score [Why?].

### Bulk



## Methods - Proof of concept

### Methods - Normally distributed simulation

### Methods - scRNA-seq pseudo-bulks

## Methods - Evaluation

### Methods - Comparison to BayesPrism

### Methods - Prediction

# Results

## Results - Proof of concept

### Results - Normally distributed simulation

### Results - scRNA-seq pseudo-bulks

## Results - Evaluation

### Results - Comparison to BayesPrism

### Results - Prediction

## Results - Proof of concept

# Discussion

## Discussion - Proof of concept

### Discussion - Normally distributed simulation

### Discussion - scRNA-seq pseudo-bulks

## Discussion - Evaluation

### Discussion - Comparison to BayesPrism

### Discussion - Prediction

## Discussion - Proof of concept

## Discussion - Outlook

# References

<div id="refs"></div>

# Supplementary Material