---
title: "Cancer transcriptome cleaning by deconvolution"
author: "Jonas Freimuth"
output:
  html_document:
    toc: true
    toc_depth: 3
bibliography: citations/cancer-cleaning_File.bib
---

```{r utils, echo = FALSE}
EQ_REFS <- list()

eq_register <- function(id) {
  # Register an equation at the point of definition and print its index.
  eq_ind <- length(EQ_REFS) + 1
  EQ_REFS[[id]] <<- eq_ind

  paste0("\\quad(", eq_ind, ")")
}

eq_reference <- function(id) {
  # Return the index of a given equation.
  paste("*(", EQ_REFS[[id]], ")*")
}
```

# Introduction

Due to its relatively low cost, bulk RNA sequencing (bulk RNA-seq) has seen
widespread adoption in clinical cancer studies [@hong.2020]. While there
currently is a boom in single-cell and spatial transcriptomics, which both allow
for highly detailed expression profiling on the level of individual cells, the
associated cost remains a factor preventing these technologies from being a
resource of large-scale, high quality data-sets. Such resources would, however,
be of great use in generating new insights in cancer research when combined with
the wealth of machine-learning applications developed over the last decades
(e.g. @garg.2021, @newell.2022). With the limitations of the more advanced
methods in mind, efforts to improve the usability of tumor bulk RNA-seq data for
machine-learning tasks would be well spent.

However, the resolution of bulk RNA-seq data compared to the other methods
mentioned above is low. For tumor samples this means that bulk RNA-seq profiles
do not represent pure expression of cancerous cells, as a tumor tissue sample
usually also includes other cell types. These may be cells belonging to the
surrounding tissue (possibly due to imprecision during biopsy), or immune cells
(like tumor infiltrating lymphocytes). These cells constitute the tumor
microenvironment (TME), whose influence on disease progression [@barkley.2022]
and as a treatment target itself [@xiao.2021] is still being studied. When
sequencing tumor tissue samples, the transcripts expressed in these non-tumor
cells become incorporated into the expression data [@li.2021], which can
represent confounding for downstream predictive modeling taking. If the
expression profiles of the TME and of the cancer cells could be separated, the
performance of such models may be increased.

Ever since the widespread adoption of bulk gene expression quantification
technologies in the form of microarrays, researchers had to address the problem
of dissecting the combined expression of mixtures of heterogeneous cells
[@venet.2001; @ghosh.2004; @stuart.2004]. Most early methods attempted only to
infer differentially expressed genes between the cell types [@ghosh.2004], but
later ones were already able to estimate cell type specific expression
[@lahdesmaki.2005], with the latter process often being called *in-silico*
purification. Historically, these methods were based on non-negative matrix
factorization (NMF), which modeled the problem as consisting of

* a $n \times p$ bulk matrix, $B$ (also called mixture matrix), giving the
  expression of $n$ genes for $p$ samples (mixtures/measurements, depending on
  context),
* a $n \times m$ signature matrix, $S$, giving the cell type specific expression
  signature of $m$ cell types,
* and a $m \times p$ concentration matrix $C$, giving the abundance of each
  cell type in each sample.

A priori, only $B$ is experimentally measured, $S$ and $C$ are estimated by
solving

$$B = S C `r eq_register("basic_matrix_deconv")`$$

with some simple constraints allowing to a tractable, unique solution
[@venet.2001]. This approach on its own, while impressive, had several
drawbacks. While not as big an issue for deconvolving only two cell types, the
resulting cell type specific estimates for abundance and gene expression could
not be directly attributed to specific cell types. The number of cell types had
to be known a priori, and misspecification would bias model results
[@lahdesmaki.2005]. Also, model solutions would depend on the set of samples,
with a low number of samples leading to worse estimates [@repsilber.2010].
@gaujoux.2012 improved the NMF approach by incorporating the prior knowledge of
genes particularly highly expressed in specific cell types. This represents the
step from an entirely unsupervised method of deconvolution, where no prior
information besides the mixture data was needed, to more accurate supervised
methods. However, previous methods only allow for estimation of the general
expression profiles of their respective cell types / conditions.

Modern high resolution methods for *in-silico* purification of bulk tumor
RNA-seq data are usually supervised full deconvolution methods. Of the 20 bulk
RNA-seq deconvolution methods reviewed in @im.2023, only three included the
capacity to estimate cell type specific expression profiles from more than two
individual samples, all of them use scRNA-seq derived reference profiles for
deconvolution. Theses methods are Cibersortx [@newman.2019], a
$\nu$-support-vector-regression ($\nu$-SVR) based general purpose method, BLADE
[@andradebarbosa.2021], a probabalistic general purpose method, and BayesPrism
[@chu.2022] a probabalistic method specifically designed for tumor
deconvolution. Cibersortx accomplishes the purification by using a multi-stage
matrix factorization approach supplemented by the cell type abundance
information estimated via $\nu$-SVR. The latter two methods' probabalistic
approaches use fairly complex model formulations and, in the case of BLADE, need
to rely on hyperparameter tuning for optimal performance. Hence, a simpler
deconvolution model might be useful in cases where explainability and
computational efficency are more important than high accuracy.

As mentioned above, the problem of pure cell type deconvolution can be modeled
as a system of linear equations `r eq_reference("basic_matrix_deconv")`.

Cell type deconvolution already exists as a way to estimate the proportions of
cell types contributing to the expression profile of a bulk tissue RNA-seq
sample. One of the basic approaches consists of solving a system of linear
equations, minimizing the distance between the transcript predictions and the
bulk data.

[High variability of cancer cells vs. remaining TME cells.]

Under the assumption that cancer cells vary stronger in their
expression than the other cells in the microenvironment, we can expect the
residuals to contain much of this excess variation, thus representing a way to
“clean” the bulk cancer expression of the expression of non-cancer cells.

[Compare to BayesPrism only as it works natively in R while BLADE is python, Cibersortx is container]


* Average expression profiles derived from scRNA-seq data
* Issue of increasing distance between actual cell expression and average
  expression profile.
* A way to possibly “boost” the signal of pure cancer expression, would be to
  not include the cancer cell type in the reference. This would of course 
  decrease the accuracy of the deconvolution model for predicting cell type
  proportions, but would in theory increase the variance in expression due to
  cancer cells that is not explained by the model, and which would therefore end
  up in the residuals. [I think however that it's not straightforward which
  portions end up in the residuals. There is still a fraction of each expression
  which gets “mis-attributed” to another cell type, but which counts as
  accounted for and thus is not present in the residuals.]

Research questions:

* Can the residuals from basic linear deconvolution methods be used to infer
  information about the tumor and inform treatment?
* Are the residuals gathered this way superior to using plain bulk data?

# Methods & Data

## Data

Single cell RNA sequencing data consists of transcript count data breast cancer
tumor samples from @wu.2021 (Gene Expression Omnibus, GEO accession
[GSE176078](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE176078)). It
contains 100'064 cells, 29'733 genes, from 26 samples. The cells are annotated
as belonging to one of nine cell types, with further sub-type information
available. A subset of 20 tumor samples has been processed via bulk RNA-seq.

For prediction tasks, bulk RNA-seq data of breast cancers annotated with
clinical data was obtained from the Cancer Genome Atalas (TCGA). The data was
comprised of all 1'095 patient samples from the TCGA-BRCA project with bulk
RNA-seq data available [Ensure reproducibility]. For the survival prediction
task, survival endpoint data derived from curated clincial data by @liu.2018 was
retrieved for the corresponding tumor samples.

## Methods - Transcriptome cleaning overview

To clean a bulk RNA-seq transcriptome, you first need to run a deconvolution on
it.

[Explanation of deconvolution as least squares problem & why nnls is sensible.]
[Mention other, more advanced methods.]
[Mathematical explanation.]
[Problem using cleaned bulk RNA-seq profiles as reference: It would need to be ensured that each profile represents a single-cell.]

For this you need a Reference and the bulk RNA-seq data to deconvolute.

### Reference

Here, the Reference is always made up averaged expression profiles derived from
scRNA data. For this, the per gene expression counts from @wu.2021 were averaged
across all cells of the same cell type. Since the dataset provided by @wu.2021
is already filtered for high quality cells, no cell quality control was
performed by me.

To explore the effects of restricting the set of genes used for deconvolution on
estimation accuracy for both cell type abundances and cancer expression, a set
of selection criteria was applied to the genes of the Reference. (First was
basic gene quality control, which comprised filtering the genes to be expressed
in at least 5 cells, removal of )

Second, each gene was assigned a marker score, quantifying its likely usefulness
as a gene in the deconvolution reference. Genes with a good score would have an
expression signature across cell types which would allow for clear
identification of just a single cell type by having particularly high or low
expression in just that cell type. Multiple ways to determine this score were
explored during preliminary experiments (DESeq2 @love.2014, Hampel filter
@davies.1993), but ultimately a simple Wilcoxon test [wilcoxon.1945] for
comparing each cell type to the pooled remaining cell types (via
`FindAllMarkers` from the R Seurat package v.5, @hao.2023) was chosen for its
combination of relative simplicity and accuracy. The marker score for the
Wilcoxon test based marker selection was the test p-Value (adjusted for multiple
testing via the Holm-Bonferroni method [@holm.1979]). Only genes differentially
expressed in a single cell type with at $p_{adj} < 0.05$ were considered as
marker genes, with lower values being considered better. It was contrasted with
choosing a random value for the marker score [Why?].

### Bulk



## Methods - Proof of concept

Before applying my method to real-world problems, I carried out two analyses
assesing its performance under ideal conditions. First, I explored the method's
behavior when applied to normally distributed data with varying levels of noise
added. Then, I quantified its performance under 

### Methods - Normally distributed simulation

To demonstrate the functionality of the principle of extracting expression
information of a heterogeneous cell type from the residuals of a linear
deconvolution model, I first applied the method to highly simplistic, normally
distributed data. 

### Methods - scRNA-seq pseudo-bulks

I then proceeded to apply the method to *in-silico* simulated bulk RNA-seq data
("pseudo-bulks") generated from the scRNA-seq data obtained from @wu.2021, to
assess its performance on actual expression data with precisely known cell type
abundances and expression profiles.

## Methods - Evaluation

After ensuring my method behaves as expected, I compared the performance to an
established method. First, I tested performance on correctly inferring true
cancer expression in both pseudo-bulks and actual RNA-sequencing data. Second, I
used cancer expression predictions of bulk RNA-seq data to train machine
learning models predicting tumor subtype and progression free interval (PFI)
and compared the performance of both BayesPrism and my method to models trained
on raw bulk data.

### Methods - Comparison to BayesPrism

To compare my transcriptome cleaning method to an established state-of-the-art
method of estimating cell type specific expression profiles from bulk RNA-seq
data, I applied both my method and BayesPrism to both tumor sample derived
pseudo-bulks, as well as to actual bulk RNA-seq samples, both from @wu.2021.

### Methods - Prediction

As the final evalutation of the method, I compared the predictive power of
machine learning models trained on both estimated cancer expression data to
models trained on bulk RNA-seq data alone.

# Results

## Results - Proof of concept

### Results - Normally distributed simulation

### Results - scRNA-seq pseudo-bulks

## Results - Evaluation

### Results - Comparison to BayesPrism

### Results - Prediction

## Results - Proof of concept

# Discussion

## Discussion - Proof of concept

### Discussion - Normally distributed simulation

### Discussion - scRNA-seq pseudo-bulks

## Discussion - Evaluation

### Discussion - Comparison to BayesPrism

### Discussion - Prediction

## Discussion - Proof of concept

## Discussion - Outlook

* Need to benchmark microarray expression purification techniques on bulk
  RNA-seq data.
* [Look into if spillover correction may be worked into method formulation.]

# References

<div id="refs"></div>

# Supplementary Material