---
title: "Cancer transcriptome cleaning by deconvolution"
author: "Jonas Freimuth"
output: html_document
bibliography: citations/cancer-cleaning_File.bib
---

```{r setup, echo = FALSE}
suppressPackageStartupMessages({
  library("yaml")
  library("dplyr")
})

suppressMessages(here::i_am("./layout.Rmd"))

rnd_deconv_params <- read_yaml(here::here(
  "cancer-cleaning-output/notebook/random_deconv_exploration_output/settings.yaml"
))

script_run_params <- read_yaml(here::here(
  "cancer-cleaning-output/script_run/reproducibility/params/merged.yaml"
))

embed_cap_plot <- function(plot_file, caption = NULL) {
  if (is.null(caption)) {
    cap_filename <- paste0(
        stringr::str_replace(plot_file, "\\.[^.]+$", ""),
        "_caption.txt"
      )
    caption <- readLines(con = cap_filename)
  }

  paste0(
    "![](", plot_file, ")\n*", caption, "*"
  )
}

embed_list_table <- function(list) {
  header <- paste(
    "| Name | Value |",
    "|-----:|:------|",
    sep = "\n"
  )

  x <- unlist(list)

  body <- paste(
    "|", names(x), "|", x, "|",
    collapse = "\n"
  )

  paste(
    header, "\n", body, collapse = "\n"
  )
}

embed_perf_table <- function(table_file) {
  data.table::fread(table_file) %>%
    select(-matches("*_name")) %>%
    select(feature_set, everything()) %>%
    mutate(across(where(is.numeric), function(x) round(x, 3))) %>%
    DT::datatable()
}
```

## Introduction

* Bulk RNA-seq has been widely adopted for many years [@Li.2021] $\rightarrow$
  There's an abundance of data available.
* Contains an unknown fraction of non-cancer expression [@Li.2021]
* This is a blokade in leveraging bulk RNA-seq data for machine learning
* If cancer expression and microenvironment expression could be separated, a
  large amount of data would be available for more accurate analysis and
  training.
* Bulk RNA-seq data can be characterized as a linear combination of average cell
  type expression plus varying levels of noise, technical and biological
  $\rightarrow$ Deconvolution via least squares

Research questions:

* Can the residuals from basic linear deconvolution methods be used to infer
  information about the tumor and inform treatment?
* Are the residuals gathered this way superior to using plain bulk data?
* Is there a difference in the usefulness of the residuals when the cancer
  reference is not included?

## Materials / Methods

### Data

* sc RNA-seq: Dataset by @Wu.2021
  * 100'064 cells, from 26 samples, 29'733 genes in total
* Clinical data: Using all available BRCA project data from TCGA (ver. 1.30.4
  see `MANIFEST.txt`)
  * Retrieved via `TCGA-biolinks` [@Colaprico.2016; @Mounir.2019]
  * Using STAR count data
  * Using sums of counts per `gene_name`!
  * Assays: Raw counts, FPKM (upper-quartile norm'd)
* Survival data: TCGA-BRCA derived survival endpoints from @Liu.2018

### Random simulation

Using normally distributed data to simulate the deconvolution process and the
relationship of bulk expression, fitted values, residual, and cancer expression.

* Done to show the method works under ideal conditions.
* Testing the effect of removing the cancer signature from the reference.
* Testing the effect of a cancer signature which is more variable than other
  cell types in the reference.

1. Generation of reference matrices of per cell type average expression
    1. Two binary categories: Hetrogen cancer signature, inclusion of cancer,
      signature $\rightarrow$ Four references in total
1. Generation of per cell bulk expression matrix from each reference matrix and
   a vector of cell type counts
1. Generation of a bulk expression vector by summing up each row of the bulk
   expression matrix, adding technical noise.
1. Deconvolution of the bulk vector using the reference matrix to obtain
   predicted cell type counts, fitted values, and residuals

Parameters used:

*Values for the biological to technical noise ratio parameter*

`r embed_list_table(rnd_deconv_params$bio_tech_noise_ratios)`

*Parameters fixed across ground truths*

| Description                                                                                  | Value                                                           |
|---------------------------------------------------------------------------------------------:|:----------------------------------------------------------------|
| Number of replicates per parameter combination.                                              | `r rnd_deconv_params$fix_params$n_repeats`                      |
| Number of cell types per ground truth.                                                       | `r rnd_deconv_params$fix_params$n_celltype`                     |
| Number of genes per ground truth.                                                            | `r rnd_deconv_params$fix_params$n_transcript`                   |
| Lower bound for number of cells per cell type.                                               | `r rnd_deconv_params$fix_params$ctype_counts_lower`             |
| Upper bound for number of cells per cell type.                                               | `r rnd_deconv_params$fix_params$ctype_counts_upper`             |
| Basic amount of noise in the reference.                                                      | `r rnd_deconv_params$fix_params$basic_noise_sd`                 |
| Relative amount of biol. noise to be added to each cell of the bulk expr. matrix.            | `r rnd_deconv_params$fix_params$cell_diff_bio_noise_rel`        |
| Relative amount of biol. noise to be added to the cancer expr. signature in the ref. matrix. | `r rnd_deconv_params$fix_params$cancer_to_transcript_noise_rel` |


Analysis:

* Four metrics
  * RMSE between true & predicted cell type counts: Basic deconv. accuracy
  * Correlation of cancer expr vs. residuals: This is the main thing we are
    interested in, it represents the degree to which residuals may be used as a
    surrogate for actual cancer expression. The higher, the better.
  * Correlation of bulk expr vs. cancer expr: Used as a baseline to
    contextualize other results. If this correlation is high, and the
    correlation of bulk expr. to residuals is high, it would also follow that
    the correlation of cancer expr. with residuals is high, but just by
    definition, not by virtue of the method.
  * Correlation of bulk expr vs. residuals: If this is high, it hints at
    problems with deconvolution. Generally deconvolution residuals should not be
    correlated with input data. Expected to be higher for reference missing the
    cancer signature.

### scRNA-seq based simulation

* Normalizing the reference and bulk probably not necessary for `nnls`
  * See @Cobos.2023 Fig. 3.

Deconvolution inputs:

* Reference
  * Averaging of cell expression belonging to one cell type
  * Using raw count matrix [Why?]
  * Marker selection [Expand on purpose of marker selection]
    * Using wilcox test
      * via Seurat (v5.0.1) [@Hao.2023]
      * using `Presto` implementation [@IlyaKorsunsky.2020]
      * previous normalization via `NormalizeData`, method relative counts
        (`RC`)
      * Using bonferroni method for p-value adjustment
      * Cutoff: Adjusted p-value under 0.05
    * Outlier distance (similar to Hampel filter)
      * Gene is defined as a marker for a cell type if its expression is more
        than three median absolute deviations away from the median _average_
        expression (i.e. this runs on the reference matrix)
      * No scaling constant (We can't assume normality, this is just arbitrary)
    * Using random genes as markers
      * Cutoff: None, at threshold 1 all genes are used.

* Pseudobulk
  * Summing up all transcripts of a tumor sample
  * Using log-normed counts (via `scuttle::logNormCounts` [@McCarthy.2017])

Deconvolution:

* Using nnls [@KatharineM.Mullen.2023] (Simple)
* _Always_ using only marker genes to estimate cell type abundances
* Transcript predictions computed (Deconvoluted cell type abundance $*$ _full_
  reference matrix) $\rightarrow$ Getting transcript predictions for _all_ genes.
* Residuals computed as pseudobulk expression - transcript predictions
  * absolute values (they are supposed to be a measure for bulk expression, and
    so can’t be negative. It’s about the deviation, the sign is not relevant)

$\rightarrow$ Resulting data is available for all genes by default, but can be subset
to only markers.

Analysis:

* Correlations computed
  * Same corr metrics as in [random simulation](#random-simulation-1)
  * All correlates log transformed (for reasonably accurrate pearson
    correlation)
  * Due to the nature of single-cell data, a large number of transcripts per
    pseudo-bulk have no associated expression, leading to zero inflated
    distributions (see [supp. plots](#scrna-seq-based-simulation-3)), where
    normal correlation measures are not meaningful. Therefore correlations are
    for this section are reported as Kendalls $\tau$ adjusted for zero
    inflation, as defined in @Pimentel.2015.

### Verification

All expression data used for training (i.e. bulk & residuals) were logscaled
before training.

#### Prediction of an intrinsic categorical variable
```{r cat-pred-params, echo=FALSE}
n_fold <- script_run_params$prediction$control$categorical_prediction$n_cv_folds
n_repeat <- script_run_params$prediction$control$categorical_prediction$n_cv_repeats
```

* PAM50 subtype
* Performing same deconvolution procedure as during pseudobulk simulation ->
  obtaining residuals as training data
* Model training
  * Using TCGA bulk data and residuals (derived from references including and
    not including cancer) as predictors
    * 1'094 Tumor samples, 59'427 transcripts in total
      * Number of transcripts varies based on feature set, see supp tables for
        details.
    * Filtered to only include "Primary solid Tumor" samples
  * Doing repeated cross-validation (`r n_fold`-fold, `r n_repeat` repeats,
    SMOTE up-sampling)
  * Pre-processing: Decorrelation via PCA, excluding zero variance genes
  * Training SVM, linear kernel
  * Using a 70-30 training-testing split
    * 765 samples for training, 329 samples for testing
* Model evaluation
  * Using model accuracy (percentage of correct predictions) to assess
    performance

#### Prediction of a clinical variable

```{r clin-pred-params, echo=FALSE}
n_tree <- script_run_params$prediction$control$survival_prediction$n_tree
```

* Progression free interval
  * Highest data quality
  * More data available due to shorter required follow-up times
* Training random survival forests [@Ishwaran.2008; @H.Ishwaran.2020]
* Using same sample data as in previous section but filtered to fit survival
  data
  * Filtered to only include "Primary solid Tumor" samples
  * Using unique samples (Multiple tumor samples from the same patient are
    present in the TCGA data.)
* Tuning model for best out-of-sample error using fast survival forest fitting
  via sub-sampling.
  * Using 200 trees
* Using optimal parameters to fit a full model.
* Using reported Harrell's C to assess performance [@Schmid.2016].

## Results

### Random simulation

`r embed_cap_plot(here::here("cancer-cleaning-output/notebook/random_deconv_exploration_output/metric_summary_plot.png"))`

### scRNA-seq based simulation

`r embed_cap_plot(here::here("cancer-cleaning-output/script_run/simulation/plots/deconv_summary/metric_summary_plot_kendall.png"))`

### Verification

#### Prediction of an intrinsic categorical variable

`r embed_cap_plot(
  here::here("cancer-cleaning-output/script_run/prediction/categorical_prediction/performance_plot.png")
)`

*Categorical prediction performance table*
`r embed_perf_table(here::here(
  "cancer-cleaning-output/script_run/prediction/categorical_prediction/performance_table.csv"
))`

#### Prediction of a clinical variable

`r embed_cap_plot(
  here::here("cancer-cleaning-output/script_run/prediction/survival_prediction/c_index_plot.png")
)`

*PFI prediction performance table*
`r embed_perf_table(here::here(
  "cancer-cleaning-output/script_run/prediction/survival_prediction/c_indices.csv"
))`

## Discussion

### Random simulation

### scRNA-seq based simulation

* Comparability of data distribution between random and scRNA-seq based deconv
* Issue of correlation between bulk expression & cancer expression

### Verification

### Outlook

* Might be sensible to try more advanced deconvolution approaches, i.e. SQUID
  [@Cobos.2023]

## References

<div id="refs"></div>

## Supplementary plots

### Random simulation

`r embed_cap_plot(here::here("cancer-cleaning-output/notebook/random_deconv_exploration_output/ground_truth_plot.png"))`

`r embed_cap_plot(here::here("cancer-cleaning-output/notebook/random_deconv_exploration_output/analysis_plot.png"))`

### scRNA-seq based simulation

`r embed_cap_plot(
  here::here(
    "cancer-cleaning-output/script_run/simulation/plots/deconv_summary/cor_FALSE_kendall_cancer_expression_v_deconv_residuals_plot.png"
  ),
  paste(
    "Rank plot of true cancer expression versus deconvolution residuals",
    "across pseudo-bulks",
    "using a reference INcluding cancer signatures",
    "facetted by whether the correlation was computed using only marker",
    "transcripts (inner rows), by the quantile of marker significance used",
    "(outer rows), and by the method used to select deconvolution markers",
    "(columns).",
    "Facet text shows the mean kendall correlation."
  )
)`

`r embed_cap_plot(
  here::here(
    "cancer-cleaning-output/script_run/simulation/plots/deconv_summary/cor_TRUE_kendall_cancer_expression_v_deconv_residuals_plot.png"
  ),
  paste(
    "Rank plot of true cancer expression versus deconvolution residuals",
    "across pseudo-bulks",
    "using a reference EXcluding cancer signatures",
    "facetted by whether the correlation was computed using only marker",
    "transcripts (inner rows), by the quantile of marker significance used",
    "(outer rows), and by the method used to select deconvolution markers",
    "(columns).",
    "Facet text shows the mean kendall correlation."
  )
)`

### Verification
