# Introduction

Bulk RNA sequencing (RNA-seq) is widely used in clinical cancer studies due to
its relatively low cost [@hong.2020]. While advanced methods like single-cell
and spatial transcriptomics provide detailed cellular-level expression
profiling, their high cost limits large-scale, high-quality data generation.
These datasets, combined with machine-learning applications
[e.g. @garg.2021; @newell.2022], could significantly enhance cancer research.
Therefore, improving the usability of tumor bulk RNA-seq data for
machine-learning tasks is crucial.

The lower resolution of bulk RNA-seq data, typically encompassing entire tissue
samples, carries the risk of incorporating non-cancerous cells, such as
surrounding tissue or tumor-infiltrating immune cells. These non-tumor cells,
which form the tumor microenvironment (TME), have a complex impact on disease
progression [@barkley.2022] and can be treatment targets themselves
[@xiao.2021]. However, their expression profiles are mixed with tumor expression
in bulk RNA sequencing [@li.2021], potentially confounding downstream predictive
modeling. Therefore, separating cancer cell expression profiles from the TME can
improve model performance.

Researchers have focused on dissecting the combined expression of heterogeneous
cell mixtures since the widespread adoption of bulk gene expression
quantification technologies, such as microarrays
[@venet.2001; @ghosh.2004; @stuart.2004]. Initially, most methods aimed to infer
only differentially expressed genes between cell types [@ghosh.2004], but later
methods could estimate cell type-specific expression [@lahdesmaki.2005], often
referred to as in-silico purification. Historically, these methods relied on
non-negative matrix factorization (NMF), which modeled the problem with the
following components:

* a $n \times p$ bulk matrix, $\mathbf{B}$ (also called mixture matrix), giving
  the expression of $n$ genes for $p$ samples (mixtures/measurements, depending
  on context),
* a $n \times m$ signature matrix, $\mathbf{S}$, giving the cell type specific
  expression signature of $m$ cell types,
* and a $m \times p$ concentration matrix $\mathbf{C}$, giving the abundance of
  each cell type in each sample.

A priori, only $\mathbf{B}$ is experimentally measured, $\mathbf{S}$ and
$\mathbf{C}$ are estimated by solving:

\begin{equation}
  \mathbf{B} = \mathbf{S} \mathbf{C}
  (\#eq:basic-matrix-deconv)
\end{equation}

This is feasible with some simple constraints that allow for a tractable, unique
solution [@venet.2001]. However, this innovative approach had several drawbacks.
While not a significant issue for deconvoluting only two cell types, the
resulting cell type-specific estimates for abundance and gene expression could
not be directly attributed to specific cell types. The number of cell types had
to be known a priori, and misspecification would bias model results
[@lahdesmaki.2005]. Additionally, model solutions depended on the set of
samples, with a low number of samples leading to poorer estimates
[@repsilber.2010]. @gaujoux.2012 improved the NMF approach by incorporating
prior knowledge of genes highly expressed in specific cell types. This
transition from entirely unsupervised deconvolution, where only mixture data was
required, to more accurate supervised methods was significant. However, these
methods only estimate the general expression profiles of their respective cell
types/conditions.

Today, most methods for *in-silico* purification of bulk tumor RNA-seq data are
supervised full deconvolution methods, but still only few methods exist for
dissecting cell type expression on the level of the individual sample
[@im.2023]. Of the 20 bulk RNA-seq deconvolution methods reviewed in @im.2023,
only three were able to estimate cell type specific expression profiles of more
than two individual samples, all of them use scRNA-seq derived reference
profiles for deconvolution. These methods are Cibersortx [@newman.2019], a
$\nu$-support-vector-regression ($\nu$-SVR) based general purpose method, BLADE
[@andradebarbosa.2021], a probabilistic general purpose method, and BayesPrism
[@chu.2022] a probabilistic method specifically designed for tumor
deconvolution. Cibersortx accomplishes the purification by using a multi-stage
matrix factorization approach supplemented by the cell type abundance
information estimated via $\nu$-SVR. The latter two methods' probabilistic
approaches use fairly complex model formulations and, in the case of BLADE, need
to rely on hyperparameter tuning for optimal performance. Hence, a simpler
deconvolution model might be useful in cases where explainability and
computational efficiency are more important than high accuracy.

<!-- Possibly expand on high cancer variability. -->

Here I introduce and examine a simple method for estimating the expression
profile of cancer cells in individual bulk RNA-seq samples by using model
residuals from simple least-squares based cell type deconvolution. Thereby, I
hope to "clean" the TME expression out from the bulk expression profile. Cell
type deconvolution is a special case of the NMF method for full deconvolution
outlined above for when the cell type specific expression profiles are already
known. Cancer cells generally show increased variability in their expression
compared to surrounding cell types [@nguyen.2016; @ramonycajal.2020]. For
least-squares based cell type deconvolution, this implies that a higher
proportion of the cancer cell's variability in expression will be present in the
deconvolution residuals compared to the other cell types. Therefore, the
residuals of such a deconvolution model fit should contain some information
specific to the expression patterns of cancer cells.

Using this approach, I will explore if the residuals from simple least squares
based cell type deconvolution actually contain information on the expression
patterns of cancerous cells in a bulk RNA-seq sample of tumor tissue. I will
further evaluate if the information used can be used to train machine learning
models which have improved predictive power compared to models trained on
ordinary bulk RNA-seq data.
