# Introduction

Due to its relatively low cost, bulk RNA sequencing (bulk RNA-seq) has seen
widespread adoption in clinical cancer studies [@hong.2020]. More advanced
methods like single-cell and spatial transcriptomics have been available for
some years, allowing for highly detailed expression profiling on a cellular
level, but the associated cost remains a factor preventing these technologies
from being a source for large-scale, high quality data-sets. Such data-sets
would, however, be of great use in generating new insights in cancer research
when combined with the wealth of machine-learning applications developed over
the last decades (e.g. @garg.2021, @newell.2022). With the limitations of the
more advanced methods in mind, efforts to improve the usability of tumor bulk
RNA-seq data for machine-learning tasks would be well spent.

However, as mentioned, the resolution of bulk RNA-seq data compared to the more
advanced methods is low. For tumor samples this implies that bulk RNA-seq
profiles do not represent pure expression of cancerous cells, since tumor tissue
samples generally also include cells of other types. These cells may be of the
surrounding tissue (possibly due to imprecision during biopsy), or immune cells
(like tumor infiltrating lymphocytes). These cells constitute the tumor
microenvironment (TME), and their influence on disease progression
[@barkley.2022] and as a treatment target itself [@xiao.2021] is still being
studied. When sequencing tumor tissue samples, the transcripts expressed in
these non-tumor cells become incorporated into the expression data [@li.2021],
which can represent confounding for downstream predictive modeling. If the
expression profiles of the cancer cells could be separated from the profile of
the TME, the performance of such models may be increased.

The problem of dissecting the combined expression of mixtures of heterogeneous
cells has been a focus for researchers since the widespread adoption of bulk
gene expression quantification technologies in the form of microarrays
[@venet.2001; @ghosh.2004; @stuart.2004]. Most early methods attempted to infer
only differentially expressed genes between the cell types [@ghosh.2004], but
later ones were already able to estimate cell type specific expression
[@lahdesmaki.2005], with the latter approach often being called *in-silico*
purification. Historically, these methods were based on non-negative matrix
factorization (NMF), which modeled the problem as consisting of

* a $n \times p$ bulk matrix, $\mathbf{B}$ (also called mixture matrix), giving
  the expression of $n$ genes for $p$ samples (mixtures/measurements, depending
  on context),
* a $n \times m$ signature matrix, $\mathbf{S}$, giving the cell type specific
  expression signature of $m$ cell types,
* and a $m \times p$ concentration matrix $\mathbf{C}$, giving the abundance of
  each cell type in each sample.

A priori, only $\mathbf{B}$ is experimentally measured, $\mathbf{S}$ and
$\mathbf{C}$ are estimated by solving

\begin{equation}
  \mathbf{B} = \mathbf{S} \mathbf{C}
  (\#eq:basic-matrix-deconv)
\end{equation}

with some simple constraints allowing to a tractable, unique solution
[@venet.2001]. This approach on its own, innovative as it was, had several
drawbacks. While not as big an issue for deconvoluting only two cell types, the
resulting cell type specific estimates for abundance and gene expression could
not be directly attributed to specific cell types. The number of cell types had
to be known a priori, and misspecification would bias model results
[@lahdesmaki.2005]. Also, model solutions would depend on the set of samples,
with a low number of samples leading to worse estimates [@repsilber.2010].
@gaujoux.2012 improved the NMF approach by incorporating the prior knowledge of
genes particularly highly expressed in specific cell types. This represents the
step from an entirely unsupervised method of deconvolution, where no prior
information besides the mixture data was needed, to more accurate supervised
methods. However, all of these methods only allow for estimation of the general
expression profiles of their respective cell types / conditions.

Today, most methods for *in-silico* purification of bulk tumor RNA-seq data are
supervised full deconvolution methods, but still only few methods exist for
dissecting cell type expression on the level of the individual sample
[@im.2023]. Of the 20 bulk RNA-seq deconvolution methods reviewed in @im.2023,
only three were able to estimate cell type specific expression profiles of more
than two individual samples, all of them use scRNA-seq derived reference
profiles for deconvolution. These methods are Cibersortx [@newman.2019], a
$\nu$-support-vector-regression ($\nu$-SVR) based general purpose method, BLADE
[@andradebarbosa.2021], a probabilistic general purpose method, and BayesPrism
[@chu.2022] a probabilistic method specifically designed for tumor
deconvolution. Cibersortx accomplishes the purification by using a multi-stage
matrix factorization approach supplemented by the cell type abundance
information estimated via $\nu$-SVR. The latter two methods' probabilistic
approaches use fairly complex model formulations and, in the case of BLADE, need
to rely on hyperparameter tuning for optimal performance. Hence, a simpler
deconvolution model might be useful in cases where explainability and
computational efficiency are more important than high accuracy.

<!-- Possibly expand on high cancer variability. -->

Here I propose a simple method for estimating the expression profile of cancer
cells in individual bulk RNA-seq samples through using model residuals from
simple least-squares based cell type deconvolution. Cell type deconvolution is a
special case of the NMF method for full deconvolution outlined above for when
the cell type specific expression profiles are already known. Cancer cells
generally show increased variability in their expression compared to surrounding
cell types [@nguyen.2016; @ramonycajal.2020]. For least-squares based cell type
deconvolution, this implies that a higher proportion of the cancer cell's
variability in expression will be present in the deconvolution residuals
compared to the other cell types. Therefore, the residuals of such a
deconvolution model fit should contain some information specific to the
expression patterns of cancer cells.

Using this approach, I will explore if the residuals from simple least squares
based cell type deconvolution actually contain information on the expression
patterns of cancerous cells in a bulk RNA-seq sample of tumor tissue. I will
further evaluate if the information used can be used to train machine learning
models which have improved predictive power compared to models trained on
ordinary bulk RNA-seq data.
