# Methods & Data

## Transcriptome cleaning overview

Following from the basic deconvolution model of equation
\@ref(eq:basic-matrix-deconv), the problem becomes tractable when the signature
matrix $\mathbf{S}$ is already known as it is then reduced to a system of linear
equations for each bulk $k$:

\begin{equation*}
  \mathbf{B}_{\cdot k} = \mathbf{S} \mathbf{C}_{\cdot k}
  \qquad \forall 1 \leq k \leq n
\end{equation*}

To reduce the solution space, usually a non-negativity constraint is placed on
elements of $\mathbf{C}$, as a mixture can't contain negative concentrations
[@venet.2001]. Adding this constraint, and considering the bulks $\mathbf{b}$
and associated concentration vectors $\mathbf{c}$ independently the problem can
be formulated as

\begin{equation*}
  \mathbf{b} = \mathbf{S} \mathbf{c}
  \qquad s.t.\ \mathbf{c}_i < 0 \; \forall 1 \leq j \leq m
\end{equation*}

This can then be solved using a least-squares method with the objective function

\begin{equation}
  \min_{\mathbf{c} \geq 0} \|\mathbf{Sc} - \mathbf{b}\|^2
\end{equation}

<!-- TODO Mention the scale on which cell type abundances can be expected to be
on & scaling considerations. -->

Under the assumption that cancer cells vary stronger in their expression than
the other cells in the microenvironment, we can expect the residuals to contain
much of this excess variation. Using these residuals as a measure of true cancer
expression thus represents a way to “clean” the bulk cancer expression of the
expression of non-cancer cells. It may be possible to further enhance the signal
of pure expression of cancer cells by simply not including the reference profile
for cancer cells during deconvolution. While this would certainly lead to less
accurate results for the prediction of cell type abundance as some proportion of
cancer cell gene expression would be erroneously attributed to other cell types,
the residuals might then more accurately reflect the patterns of cancer cell
type expression as when they are already accounted for by the inclusion of the
cancer cell type reference profile.

To clean a bulk RNA-seq transcriptome, you first need to run a deconvolution on
it.

<!-- Problem using cleaned bulk RNA-seq profiles as reference: It would need to
be ensured that each profile represents a single-cell. -->

For this you need a Reference and the bulk RNA-seq data to deconvolute.

## Data

Prior information on the expression profiles of the cell types present in a set
of bulk RNA-seq samples may be obtained in different ways. Relatively simple
approaches like expression profiling of cultured purified cell lines allow for
characterization of cell type specific expression profiles at low cost and
effort, but require normalization to eliminate bias from differing cell counts.
More complex methods allowing for single cell level expression quantification
like micro-dissection, fluorescence-activated cell sorting, and more recently
scRNA-seq do not have this problem, but are limited in application due to their
cost. Here, I used single cell RNA sequencing data consisting of transcript count
data breast cancer tumor samples from @wu.2021 (Gene Expression Omnibus, GEO
accession
[GSE176078](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE176078)). It
contains 100'064 cells, 29'733 genes, from 26 samples. The cells are annotated
as belonging to one of nine cell types, with further subtype information
available. A subset of 20 tumor samples has been processed via bulk RNA-seq.

For prediction tasks, I used bulk RNA-seq data of breast cancers annotated with
clinical data from the Cancer Genome Atlas (TCGA). The data comprised all 1'095
patient samples from the TCGA-BRCA project with bulk RNA-seq data available
[Ensure reproducibility]. For the survival prediction task, I retrieved survival
endpoint data derived from curated clinical data by @liu.2018 for the
corresponding tumor samples.

### Reference

Here, the Reference is always made up averaged expression profiles derived from
scRNA data. For this, the per gene expression counts from @wu.2021 were averaged
across all cells of the same cell type. Since the dataset provided by @wu.2021
is already filtered for high quality cells, no cell quality control was
performed by me.

To explore the effects of restricting the set of genes used for deconvolution on
estimation accuracy for both cell type abundances and cancer expression, a set
of selection criteria was applied to the genes of the Reference. (First was
basic gene quality control, which comprised filtering the genes to be expressed
in at least 5 cells, removal of [...])

Second, each gene was assigned a marker score, quantifying its likely usefulness
as a gene in the deconvolution reference. Genes with a good score would have an
expression signature across cell types which would allow for clear
identification of just a single cell type by having particularly high or low
expression in just that cell type. Multiple ways to determine this score were
explored during preliminary experiments (DESeq2 @love.2014, Hampel filter
@davies.1993), but ultimately a simple Wilcoxon test [wilcoxon.1945] for
comparing each cell type to the pooled remaining cell types (via
`FindAllMarkers` from the R Seurat package v.5, @hao.2023) was chosen for its
combination of relative simplicity and accuracy. The marker score for the
Wilcoxon test based marker selection was the test p-Value (adjusted for multiple
testing via the Holm-Bonferroni method [@holm.1979]). Only genes differentially
expressed in a single cell type with at $p_{adj} < 0.05$ were considered as
marker genes, with lower values being considered better. It was contrasted with
choosing a random value for the marker score [Why?].

### Bulk

## Proof of concept

Before applying my method to real-world problems, I carried out two analyses
assessing its performance under ideal conditions. First, I explored the method's
behavior when applied to normally distributed data with varying levels of noise
added. Then, I quantified its performance under 

### Normally distributed simulation

To demonstrate the functionality of the principle of extracting expression
information of a heterogeneous cell type from the residuals of a linear
deconvolution model, I first applied the method to highly simplistic, normally
distributed data.

### scRNA-seq pseudo-bulks

I then proceeded to apply the method to *in-silico* simulated bulk RNA-seq data
("pseudo-bulks") generated from the scRNA-seq data obtained from @wu.2021, to
assess its performance on actual expression data with precisely known cell type
abundances and expression profiles.

I computed reference profiles across a spectrum of parameters, one reference for
each combination. I used both marker selection strategies outlined above
(Wilcoxon test, random score), each with a threshold of the top XXX of the best
scoring marker genes being selected. I also contrasted across including the
cancer cell type or not, and running the transcript quality control or not.

[Were there any references that were removed due to NULL / no markers?]

I generated XXX pseudo-bulks by randomly sampling XXX cells (twice the
median number of cells per sample in the data) with replacement from the
overall count matrix from @wu.2021, stratified by cell type. Cell type fractions
were also randomly selected by first choosing each cell type's fraction from
$[0, 1]$ uniformly at random, and then normalizing the fraction of all
non-cancer cell types to the complement of the fraction of the cancer cell type.
This was to allow for a broad range of cancer cell type fractions, which would
have been restricted to lower values due to the normalization.

I then deconvoluted each pseudo-bulk using each reference and calculated
performance metrics for estimation of both cell type concentrations and cancer
cell type expression from residuals.

[...]

## Evaluation

After ensuring my method behaves as expected, I compared the performance to an
established method. First, I tested performance on correctly inferring true
cancer expression in both pseudo-bulks and actual RNA-sequencing data. Second, I
used cancer expression predictions of bulk RNA-seq data to train machine
learning models predicting tumor subtype and progression free interval (PFI)
and compared the performance of both BayesPrism and my method to models trained
on raw bulk data.

### Comparison to BayesPrism

<!-- Compare to BayesPrism only as it works natively in R while BLADE is python,
Cibersortx is container -->

To compare my transcriptome cleaning method to an established state-of-the-art
method of estimating cell type specific expression profiles from bulk RNA-seq
data, I applied both my method and BayesPrism to both tumor sample derived
pseudo-bulks, and to actual bulk RNA-seq samples, both from @wu.2021.

### Prediction

As the final evaluation of the method, I compared the predictive power of
machine learning models trained on both estimated cancer expression data to
models trained on bulk RNA-seq data alone.