# Results

```{r res-setup, include = FALSE}
import::from("data.table", "fread", "uniqueN")
import::from("dplyr", "arrange", "mutate", "recode", "rename", "select")
import::from("here", "here")
import::from("knitr", "include_graphics", "kable")
import::from(
  "kableExtra",
  "add_header_above",
  "collapse_rows",
  "kable_minimal",
  "kable_styling",
  "kbl",
  "landscape"
)
import::from("magrittr", "%>%", "extract", "multiply_by", "set_names")
import::from("purrr", "map", "reduce")
import::from("tidyr", "pivot_wider")

# Suppress chunk output by default.
knitr::opts_chunk$set(echo = FALSE)

here::i_am("bookdown/03-results.Rmd")

analysis_root <- "cancer-cleaning-output"
main_plot_dir <- "main_plots"
main_plot_path <- here(paste(analysis_root, main_plot_dir, sep = "/"))

plot_prefixes <- list(
  "rnd_sim" = "random_deconv_simulation_metric_summary_plot",
  "pbulk_ctype_acc" = "simulation_celltype_accuracy_plot",
  "pbulk_cexpr_acc" = "simulation_true_v_predict_cancer_corr_plot",
  "bp_cexpr_acc" = "bayes_prism_true_v_predict_cancer_corr_plot",
  "pred_cat_perf" = "categorical_prediction_performance_plot",
  "pred_srv_perf" = "survival_prediction_performance_plot"
)

plot_paths <- map(
  plot_prefixes,
  \(prefix) paste(main_plot_path, paste0(prefix, ".png"), sep = "/")
)

plot_caps <- map(
  plot_prefixes,
  \(prefix) paste(main_plot_path, paste0(prefix, "_caption.txt"), sep = "/")
) %>%
  map(readLines)

plot_data_paths <- map(
  plot_prefixes,
  \(prefix) paste(main_plot_path, paste0(prefix, ".csv"), sep = "/")
)

# TODO Add table captions.
```

## Proof of concept

### Normally distributed simulation

During simulation using normally distributed data, my method's performance
behaved as expected.

<!-- TODO Rephrase this as groups having more technical noise. -->

<!-- FIXME Do I need to talk about cell type estimation accuracy at all? Is
mentioning the drop due to cancer splitting & pointing at supplementary enough?
-->

The accuracy in estimating cell type abundance was optimal for high & equal
biological noise using a reference with the cancer profile still included, and
assuming that cancer expression varies across cells similar to other cell types
(Figure \@ref(fig:rnd-sim-plt) A). Cell type estimation accuracy fell slightly
when replacing the latter assumption with a stronger variation for cancer
expression across cells, but particularly strongly for the groups of high &
equal amounts of biological noise. This left the former groups closer in
accuracy to the low biological noise group. When removing the cancer expression
profile from the reference, the estimation further decreased, with differences
between all levels of relative biological noise except the very lowest becoming
very small to non-existent, even across models for cancer expression
variability. Lastly, the accuracy for estimated cell types was invariably the
lowest for simulations with a relatively very low amount of biological noise and
barely varied in its absolute value across combinations of the remaining
parameters.

The accuracy of deconvolution model residuals as an estimate of cancer
expression on the other hand shows an almost complementary picture (Figure
\@ref(fig:rnd-sim-plt) B). Here, the correlation under the homogeneous cancer
expression and including cancer profiles in the reference was very poor.
However, there were still differences between the groups: High & equal
biological noise had the same levels of accuracy, while the low biological noise
group was slightly worse, and the very low biological noise showed levels of
accuracy barely distinguishable from zero. Under the model of cancer cells
varying stronger than other cell type's cells, and still including cancer
profiles in the reference, there was an appreciable improvement in accuracy
compared to the homogeneous model. The performance across groups remained
similar to the previous model. The best improvement can be seen however when
removing the cancer expression profile from the reference. Cancer expression
accuracy here is very good across levels of biological noise except the lowest,
and regardless of cancer expression model.

```{r rnd-sim-plt, fig.cap = plot_caps$rnd_sim}
include_graphics(plot_paths$rnd_sim)
```

```{r rnd-sim-res-tab}
rnd_sim_data <- plot_data_paths$rnd_sim %>%
  fread(sep = ",") %>%
  mutate(across(
    where(is.character),
    \(char_col) factor(char_col, levels = unique(char_col))
  ))

# Recode types so they appear in order when sorting cols after pivot_longer.
type_recode <- list(
  # Currently unused.
  "True vs. predicted Cell type abundance 1 / RMSE" = "a ctype_abundances",
  "Cancer expression vs. Model residuals R²" = "b cancer_accuracy"
)

# Depends on the same ordering as type_recode.
type_headers <- c(
  # Currently unused.
  "True vs. predicted Cell type abundance 1 / RMSE" = "Cell type accurracy, $1 / RMSE$",
  "Cancer expression vs. Model residuals R²" = "Cancer expression accurracy $\\rho$"
) %>%
  extract(names(.) %in% unique(rnd_sim_data$type)) %>%
  # Create a list with previous values as names and `2` (fixed width, mean & ci)
  # as values.
  {
    set_names(rep(list(2), length(.)), .)
  }

rnd_sim_data %>%
  # Maybe the cols removed here will be removed somewhere upriver already. If
  # that's the case, this can be removed.
  select(-any_of(c("n", "n_rmse"))) %>%
  mutate(
    `95% CI` = paste("\U00B1", round(mean - min_ci, 3)),
    type = recode(type, !!!type_recode)
  ) %>%
  select(-matches("_ci")) %>%
  pivot_wider(
    values_from = c("mean", "95% CI"),
    names_from = "type",
    names_glue = "{type} {.value}"
  ) %>%
  select(
    heterogeneous_cancer,
    split,
    bio_tech_ratio,
    # Should in ctype acc first and mean first
    all_of(sort(names(.), decr = TRUE))
  ) %>%
  arrange(heterogeneous_cancer, split, bio_tech_ratio) %>%
  kbl(
    caption = "TODO ADD PROPER TABLE CAPTIONS",
    escape = TRUE,
    col.names = c(
      "Cancer model",
      "Reference type",
      "Relative noise level name",
      rep(c("Mean", "95% CI"), uniqueN(rnd_sim_data$type))
    ),
    align = "l"
  ) %>%
  kable_minimal() %>%
  add_header_above(c(
    " " = 3,
    type_headers
  )) %>%
  collapse_rows(columns = 1:3, valign = "top") %>%
  landscape()
```

### scRNA-seq pseudo-bulks

When benchmarking my method on pseudo-bulks derived from scRNA-seq data,
the performance was not as consistent as during the simulation using normally
distributed data.

<!-- TODO Remove transcript QC from plots and analysis. As far as I can tell
it's not relevant. -->

When including the cancer expression profile in the reference, cell type
estimation accuracy was rather good, but only when a reference with randomly
chosen marker genes was used. Under these conditions I observed a steady
improvement in cell type estimation accuracy with increasing marker set size,
even though no individual step was different from the previous in a
statistically significant way (overlapping 95% confidence intervals).

```{r pbulk-ctype-acc-plt, fig.cap = plot_caps$pbulk_ctype_acc}
include_graphics(plot_paths$pbulk_ctype_acc)
```

```{r pbulk-ctype-acc-tab}
plot_data_paths$pbulk_ctype_acc %>%
  fread(sep = ",") %>%
  kable(caption = "TODO ADD PROPER TABLE CAPTIONS", escape = TRUE)
```

```{r pbulk-cexpr-acc-plt, fig.cap = plot_caps$pbulk_cexpr_acc}
include_graphics(plot_paths$pbulk_cexpr_acc)
```

```{r pbulk-cexpr-acc-tab}
plot_data_paths$pbulk_cexpr_acc %>%
  fread(sep = ",") %>%
  kable(caption = "TODO ADD PROPER TABLE CAPTIONS", escape = TRUE)
```

## Evaluation

### Comparison to BayesPrism

<!-- TODO Mention the number of genes on which comparisons are based for
BayesPrism. -->

```{r bp-cexpr-acc-plt, fig.cap = plot_caps$bp_cexpr_acc}
include_graphics(plot_paths$bp_cexpr_acc)
```

```{r bp-cexpr-acc-plt-tab}
plot_data_paths$bp_cexpr_acc %>%
  fread(sep = ",") %>%
  kable(caption = "TODO ADD PROPER TABLE CAPTIONS", escape = TRUE)
```

### Prediction

```{r pred-cat-perf-plt, fig.cap = plot_caps$pred_cat_perf}
include_graphics(plot_paths$pred_cat_perf)
```

```{r pred-cat-perf-tab}
plot_data_paths$pred_cat_perf %>%
  fread(sep = ",") %>%
  kable(caption = "TODO ADD PROPER TABLE CAPTIONS", escape = TRUE)
```

```{r pred-srv-perf-plt, fig.cap = plot_caps$pred_srv_perf}
include_graphics(plot_paths$pred_srv_perf)
```

```{r pred-srv-perf-tab}
plot_data_paths$pred_srv_perf %>%
  fread(sep = ",") %>%
  kable(caption = "TODO ADD PROPER TABLE CAPTIONS", escape = TRUE)
```
