# Discussion

## Proof of concept

### Normally distributed simulation

The experiment of applying deconvPure to normally distributed random data
demonstrated the theoretical potential of the method for estimating cancer
expression from bulk RNA-seq data.

It showed that indeed, higher variability in one cell-type would lead to
information about the expression of that cell type to be present in the
residuals of a linear deconvolution model applied to the bulk mixture. Further,
it demonstrated that entirely excluding a cell type from the reference profiles
would strongly boost that signal, an effect that is robust to relatively high
amounts of obscuring technical noise. The inherent variability of cells also
appears to only moderately impact estimation accuracy, hinting at an ability of
deconvPure to produce accurate estimates even when the cell types in a tissue
are in different phases of their cell cycle. Lastly, the random simulation
experiment shows that deconvPure's expression estimates contain more information
about the cell type's true expression than is inherently present in the bulk
expression profile.

### scRNA-seq pseudo-bulks

The high estimation accuracy achieved by deconvPure when applied to pseudo-bulks
derived from scRNA-seq data looks very promising. However, the minuteness of the
difference in estimation accuracy between including the cancer profile in the
reference or not, is not in line with the behavior expected from the previous
simulation experiments. The high correlations achieved across parameter
combinations are even more surprising when considering the small amount of
marker genes that were used to estimate cell type counts during some
deconvolutions (Supp. Table \@ref(tab:pbulk-cexpr-acc-tab)). One explanation for
the unexpectedly high estimation accuracy could simply be a very high
variability of cancer expression across cells. As the previous simulation
experiment showed, this has the potential to induce correlation between
residuals and cancer expression, at least on normally distributed data.

Another explanation for the unexpectedly high estimation accuracy, regardless of
the reference type used, and barely influenced by the amount of marker genes
used during deconvolution, could be the relationship of cell type expression
profiles among each other. As seen in Supp. Figure
\@ref(fig:rnd-sim-ground-truth) (D), during random simulation there was
essentially no co-linearity between cell types. This is not the case, however,
for the pseudo-bulks (Supp. Figure \@ref(fig:pbulk-sim-diag-1) (D) &
\@ref(fig:pbulk-sim-diag-2) (D)), where the large amounts of mutual correlation
between cell types may be responsible for the high estimation accuracy. The high
co-linearity observed for the pseudobulks could also be the driver of the high
amount of information on cancer expression present in the bulk (see below).

The absence of a significant effect of the type of marker selection methods used
on estimation accuracy, especially when excluding the cancer profile from the
reference, is unexpected but might indicate a flaw in the comparison of marker
methods. For an accurate comparison, the same amount of marker genes should be
used for both random and Wilcoxon based references. As it is, the relatively
high amount of markers for random selection compared to Wilcoxon test based
markers (Supp. Table \@ref(tab:pbulk-cexpr-acc-tab)), confounds the effect of
the marker selection method with the absolute amount of markers. This makes
inference about whether the method of selection provides a benefit difficult.
The observed lack of difference in the selection method used could be due to an
actual lack of effect, or due to the effect of marker selection being canceled
out by an effect of sheer marker set size.

Alternatively, it may just be important that certain genes are part of the
reference, which has a high probability of being the case for the large amounts
of marker genes used for random marker selection. In turn, this effect may also
explain the high variability observed for Wilcoxon markers. As mentioned, the
number of Wilcoxon based marker genes in total was rather low (I suspect this
was primarily due to the requirement that marker genes need to have differential
expression in one cell type only). This can lead to variable estimation accuracy
if those marker genes are not consistently present across samples, or be subject
to variation in expression or detection themselves.

The slightly inconsistent pattern of average cancer expression estimates across
thresholds could be an artifact of the method of ordering marker genes I used. I
designated the $p$-value as the metric, and lower values as more optimal, but
that may be not entirely appropriate. @avilacobos.2020 rather used the
fold-change of the gene expression between the cells belonging to a type and the
remaining types as the metric determining the optimal markger genes. When
ordering by $p$-value, the optimal marker genes by the fold-change metric could
already be removed at tighter thresholds, leading to unexpectedly low
performance.

The inferior correlation of deconvPure's cancer expression estimates with actual
cancer expression compared to simple bulk expression adds important context to
the absolute estimation accuracy. deconvPure, as it is, is not able to provide a
general benefit for inferring cancer expression profiles from bulk RNA-seq data.
The slight benefit conferred by a particular set of marker genes on samples with
low tumor purities seems to suggest there is some merit to the method still.
Considering the variability of estimation accuracy, however, this seems too
tenuous to be useful in any real-world application, especially when considering
the artificial nature of the data on which this was observed.


## Evaluation

### Comparison to BayesPrism

Considering its simplicity, deconvPure's performance on matched pseudo-bulk and
bulk data in absolute numbers compared to BayesPrism was rather good, especially
on bulk data. The slightly lower average estimation accuracy that deconvPure
achieved on tumor sample derived pseudo-bulks, rather than pseudo-bulks with
randomly chosen tumor purities, may be partly due to the smaller range of tumor
purities covered by the former pseudo-bulk construction method. As the previous
experiments have shown, deconvPure's estimation accuracy is associated with
estimation accuracy of cancer expression, so it is expected that running the
method on samples with generally lower tumor purity will produce slightly worse
estimation accuracy.

The performances of BayesPrism on bulk data using random markers is in line with
what @chu.2022 found when benchmarking the method on pseudo-bulks. (Although
curiously, their pseudo-bulk data derived from glioblastoma appears to not have
as much inherent information for lower tumor purities than the breast cancer
data examined here does.) The apparent low and variable performance on
BayesPrism on pseudo-bulk data using Wilcoxon based markers may be partly due to
the method not allowing for expression estimation of non-marker genes. As shown
in Supp. Table \@ref(tab:bp-cexpr-acc-plt-tab), the number of genes on which the
correlation measures are based are very different, with the small number
probably causing instability for BayesPrism.

The lower performance of both methods on bulk RNA-seq data compared to pseudo-
bulks is expected to a degree, as there is additional noise present in bulk data
compared to pseudo-bulks, such as different sample preparation. In pseudo-bulks
on the other hand, technical noise shouldn't be a factor impacting the
estimation accuracy as mixtures are created by simple addition of cell
expression. Using an appropriate reference is both one of the most difficult and
important steps in leas-squares based deconvolution [@qiao.2012]. As outlined in
the [methods](#transcriptome-cleaning-overview), the reference profile needs to
represent a single cell, necessitating the use of scRNA-seq data. This however
introduces its own method specific biases relevant for devonvolution, such as
differences in tissue dissociation across cell types [@lambrechts.2018].

An additional source of error for estimation accuracies on bulk data is the true
value used for the computation of correlations itself. Due to the differences in
general expression characteristic mentioned above, using scRNA-seq data for the
purpose of inferring true cancer expression profiles of bulk data may result in
underestimation, but it is the best thing that was close to hand. If this is
revisited, I believe repeating this comparison using true cancer expression data
from cell sorted bulk RNA-seq experiments would lead to higher accuracy.

The issue of the compromised comparability of the marker selection methods also
exists here. However, since there seems to be no improvement in using
thresholded marker genes over just all genes passing quality control (random
markers at a threshold of one), the lack of comparability of the selection
methods appears to not be particularly relevant.

deconvPure's inability to estimate cancer expression with higher accuracy than
the inherent information in bulk profiles is already expected from the previous
experiment. The almost complete lack of samples on which it managed to beat out
the inherent bulk information adds further explanation to the lower overall
accuracy. It is not only due to the smaller range of tumor purities covered, but
also due to the data examined. That BayesPrism did achieve higher estimation
accuracy than inherent information, seems to indicate that the simple approach
that deconvPure represents is not sufficient to address the challenges that
accurately decomposing the bulk expression profile into cell type specific
profiles poses.

### Prediction

The prediction of a tumor sample's PAM50 subtype did not produce results as
clear as expected. The range of the average model accuracies is rather narrow,
indicating that using estimated cancer expression or raw bulk data does not
matter either way. From the overall poor prediction accuracy for PFI, it is
clear that this is a challenging outcome to predict. deconvPure was almost
entirely unable to help in predicting PFI, even counterproductive when
considering the considerably higher accuracy of the model trained with matched
bulk data. BayesPrism's prediction leading to a slightly better model for
predicting PFI could still be an indication that an improvement in prediction
accuracy by using estimated cancer expression is possible at all. However, when
considering the full picture with marker selection, the lack of a general
pattern of this, makes this outcome seem rather to be due to chance (Supp.
Figure \@ref(fig:pred-srv-perf-supp-plt), Supp. Table
\@ref(tab:pred-srv-perf-tab)).

Considering how similar accuracies of models predicting cancer subtype from both
estimated cancer expression and matched bulk data were, it seems unlikely that
using estimated cancer expression provides a clear benefit over simply using
bulk expression data. Nevertheless, the gene set used for model training appears
to have an influence on the prediction accuracy, with the gene set used by
BayesPrism being optimal, at least for predicting cancer subtype. The
differences in the gene set between the two methods were due to BayesPrism not
being able to produce expression estimates for genes removed during the quality
control for deconvolution. Perhaps the genes removed are also detrimental to the
prediction accuracy.

The similar prediction accuracy between models trained on estimated cancer
expression and bulk expression on both prediction tasks may be explained in
large part by the inherent information of bulk expression on cancer profiles. If
both sets of features contain information that is highly correlated, a large
difference in predictive power can't be expected. Given the inability of
deconvPure to predict the cancer expression profile of a bulk with higher
accuracy than inherent information seen in previous sections, the lower
predictive ability of the models is not surprising. The fact that BayesPrism did
manage to improve model accuracy also fits with the superior estimation accuracy
demonstrated on bulk data during the
[comparison section](#comparison-to-bayesprism). However, a consistent
improvement in accuracy over matched bulk data would be expected then, but this
is not how the performance across thresholds on marker selection behaved (Supp.
Figures \@ref(fig:pred-cat-perf-supp-plt) and
\@ref(fig:pred-srv-perf-supp-plt)).

## Conclusion

<!-- If one has to choose BayesPrism works pretty well on these data when using
all data possible. Improvement on bulk data not that good actually. Seems really
good method is needed for actual improvements. -->

The principle behind deconvPure is sound and capable of inferring the expression
profile of a cell type from deconvolution residuals when excluding that cell
type's profile from the reference. The method has been benchmarked on pseudo-
bulk data and achieved high accuracy across a wide range of tumor purities.
However, while deconvPure has shown itself to be capable of reasonable
estimation accuracy on cancer expression profiles from bulk data, it could not
demonstrate an improvement over the inherent information in bulk expression. As
a consequence, I could not demonstrate a clear improvement in prediction
accuracy for models using cancer expression profiles estimated by deconvPure.

Since deconvPure is a very simple method, it has multiple straightforward
avenues of further improving its estimation accuracy. The decrease in cell type
accuracy due to the absence of the cancer reference profile could perhaps be
mitigated by using a two-step approach: First estimating cell type proportions
accurately, possibly with some other method better able to account for the
specific challenges of cell type deconvolution, and then using the same approach
of constructing "fitted values" using reference profiles without including the
cancer reference. From that a form of residuals could be computed again and used
as an estimate of cancer expression. Alternatively, the approach used in xCell
[@aran.2017] of correcting cell type estimates by known "spillover" propensities
could mitigate the effect of cancer expression being misattributed to other cell
types and thus leading to selective underestimation of cancer expression. The
best improvement could probably be achieved by simply using as accurate a
reference as can be obtained, for example by replicating experimental conditions
when re-analyzing bulk RNA-seq samples.

Further study might include exploring these improvements to deconvPure alongside
state-of-the-art methods answering actual biological questions. Also, the
estimation accuracy so far has only been assessed on breast cancer data, as I
deemed it out of the scope of this work to find other cancer scRNA-seq datasets.
Perhaps other cancer types have characteristics that are particularly favorable
or challenging for this approach.
