# Discussion

## Proof of concept

### Normally distributed simulation

The experiment of applying deconvPure to normally distributed simulated data
demonstrated the theoretical potential of the method for estimating cancer
expression from bulk RNA-seq data. It showed that indeed, higher variability in
one simulated celltype would lead to information about the expression of that
cell type to be found in the residuals of a linear deconvolution model applied
to the mixutre. Further, it demonstrated that entirely excluding a cell type
from the reference profiles would strongly boost that signal to high
correlations. That boosting effect can be observed even when adding relatively
high amounts of obscuring technical noise. Also, variability of cells only
moderately reduces estimation accuracy.

### scRNA-seq pseudo-bulks

<!-- TODO Discuss relatively high variability for Wilcoxon markers. -->

When applying deconvPure to pseudo-bulks derived from scRNA-seq data, it was
able to achieve high to very high estimation accuracy. This looks very
promising, but the minuteness of the difference in estimation accuracy between
including the cancer profile in the reference or not is not in line with the
behavior expected from the previous simulation experiments. The high
correlations achieved across parameter combinations are even more surprising
when considering the small amount of marker genes that were used to estimate
cell type counts during some deconvolutions (as low as only 62 for the tightest
threshold using Wilcoxon based markers; Table \@ref(tab:pbulk-cexpr-acc-tab)).
Also, the type of marker selection methods used (random vs. Wilcoxon-test based)
doesn't cause a particularly large difference in estimation accuracy, especially
when excluding the cancer profile from the reference.

<!-- TODO Expand on why this leads to residuals correlating with cancer even
when including cancer profiles. -->

The unexpectedly high estimation accuracy, barely influenced by the amount of
marker genes used during deconvolution, could be explained by the relationship
of cell type expression profiles among each other. As seen in Supp. Figure
\@ref(fig:rnd-sim-ground-truth) (D), during random simulation there was
essentially no co-linearity between cell types. This is not the case, however,
for the pseudo-bulks (Supp. Figure \@ref(fig:pbulk-sim-diag-1) (D) &
\@ref(fig:pbulk-sim-diag-2) (D)).

<!-- TODO Discuss acc v prop relationship. -->


## Evaluation

### Comparison to BayesPrism

* BayesPrism variability due to lower number of actual transcripts used for cor
  computation.
* Worse estimation accuracy on bulk partially due to low accuracy of true values
  of expression used.
* If this is revisited, finding a proper test dataset using matched flow
  cytometry data would be sensible.

### Prediction

#### PAM50

* PAM50 subtype can't be predicted from estimated expression with higher
  accuracy than from bulk data alone.
* Lower performance on full bulk data due to lack of QC.

#### PFI

* Already not that easy to estimate PFI.
<!-- TODO Check novelty of that. -->
* Important finding: Targeted marker selection significantly improves
  prediction for PFI.
* High co-linearity could partially explain why estimated cancer expression and
  raw bulk predictions have such similar performance.

* Prediction attempts maybe a bit crude, a more meaningful prediction task
  could give a better picture of the capability of the method.

## Conclusion

* Answers to the questions:
  * Accuracy of estimation of cancer expression from bulk data
    * The modeling approach is theoretically capable of estimating cancer
      expression at high accuracies.
    * For the purpose of doing this on bulk RNA-seq data, it seems that the
      accuracy is not worth it as much of the information is already present
      in the bulk, and the estimation resolution is not good enough to tease
      out the finer differences.
  * Improvement in model training
    * No clear improvement could be observed, in some cases expression estimates
      led to better model performance than either full bulk data or bulk data
      using the same gene-set, but never at the global optimum.

* Implications of colinerarity.
  * Again points towards high estimation possbile, but perfect corelation very
    hard.

* Possible deconvolution model improvements:
  * Spillover correction
  * Using high accuracy methods available for cell type deconvolution, use
    single-cell derived reference to compute estimated expression profile of
    TME, and calculate residuals from that.
  * Importance to get as close a reference as possible, reducing noise from
    distance between reference profile and actual cells.
  * Following from improved reference: Higher resolution reference (cell,
    subtype, cell state), like @chu.2022 did.

* All in all:
  * High accuracy estimation is possible, but not so high that prediction can be
    improved beyond the already high information content. This approach so far
    has shown good results for estimation of cancer expression.
