---
title: "Cancer Transcriptome Cleaning by Deconvolution"
subtitle: "Thesis for M.Sc. Bioinformatics at the University of Potsdam"
author: "Jonas Freimuth (807218)"
documentclass: article
output:
  bookdown::html_document2:
    toc: true
    toc_depth: 3
  bookdown::pdf_document2:
    toc: false
    toc_depth: 3
    latex_engine: xelatex
bibliography: ../citations/cancer-cleaning_File.bib
---

\vspace*{\fill}

## Revievers {.unlisted .unnumbered}

First reviewer: Dr. **Altuna Akalin**, Berlin Institute for Medical Systems
Biology, Max-Delbrück-Centrum für Molekulare Medizin,
email:&nbsp;*Altuna.Akalin@mdc-berlin.de*

Second reviewer: Prof. Dr. **Zoran Nikoloski**, Institute for Biochemistry and
Biology, University of Potsdam, email:&nbsp;*zniko@uni-potsdam.de*

\newpage

## Declaration of Originality {.unnumbered}

I, Jonas Freimuth, hereby delcare,

* that this thesis is the product of my own work. All the assistance received in
  preparing this thesis and the sources used have been acknowledged.
* All statements, literal or paraphrased, which have been taken from other works
  have been marked as such,
* and that this work is not in its entirety, nor in substantial parts, part of
  another examination procedure, nor has it been previously published.

Potsdam, the `r format(Sys.Date(), "%d. of %B %Y")`

\newpage

## Aknowledgements {.unnumbered}

I would like to thank my supervisor, Dr. Altuna Akalin for his support and
guidance throughout this thesis, as well as for the generous amounts of
computation capacity I could access for the thesis. I also extend my thanks to
Prof. Dr. Zoran Nikoloski for agreeing to review this work. I want to thank the
members of the Akalin lab for making the long commute to Berlin worth it, and
for providing important suggestions when I was stuck on particularly annoying
problems and dispelling doubts about my work. I also wish to thank my friends
for keeping me sane through the darker parts of my journey. Finally, I want to
thank my family very much for supporting me through all these years of study,
both financially and emotionally. I could not have gotten here without any of
them.

\- Jonas

\newpage

## Abstract {.unnumbered}

Bulk RNA-sequencing (bulk RNA-seq) is now a widely adopted method for
quantifying gene expression of biological samples, including in cancer research.
With more advanced but expensive methods becoming available, today bulk RNA-seq
represents a trade-off between cost and resolution, as expression profiles can't
be attributed to individual cell types. If pure expression of cancer cells is of
interest, the expression of other cell types in the sample (the tumor
microenvironment, TME) can obscure the cancer expression signal. The wealth of
existing tumor bulk expression data could be used even more effectively if it
were possible to reduce or remove this noise. I explored the feasibility of
using simple, least-squares based cell type deconvolution to estimate pure
cancer expression from bulk RNA-seq data of tumor samples. By intentionally
mis-specifying the deconvolution model, I hoped to enrich the cancer expression
signal in the model residuals. In simulation experiments I was able to
demonstrate that this works for normally distributed data and is tolerant to
simulated technical noise. When applied to simulated bulk data, I still achieved
high estimation accuracy, but was not able to show clear improvement over the
information already present in bulk data. The same was true for real-world bulk
RNA-seq data. When attempting to use cancer expression estimates to predict a
tumor- and a patient-level variable, I did not observe improvement in accuracy
over the raw bulk data. Due to its simplicity, the method still has ways to
improve and become useful for cancer research.

\newpage

## Zusammenfassung {.unnumbered}

Die Bulk-RNA-Sequenzierung ist heute eine weit verbreitete Methode zur
Quantifizierung der Genexpression in biologischen Proben, unter anderem in der
Krebsforschung. Sie stellt einen Kompromiss zwischen Kosten und Auflösung dar,
da Expressionsprofile nicht einzelnen Zelltypen zugeordnet werden können. Wenn
allerdings die reine Expression von Krebszellen für eine Fragestellung relevant
ist, kann die Expression anderer Zelltypen (der Tumormikroumgebung, TME) das
Krebsexpressionssignal abschwächen. Die Fülle vorhandener Bulk-Expressionsdaten
aus Tumorproben könnte noch effektiver genutzt werden, wenn es möglich wäre,
dieses Rauschen zu reduzieren oder zu entfernen. Hier untersuche ich die
Möglichkeit, mithilfe einfacher Zelltyp-Dekonvolution die Expressionsprofile der
Krebszellen in Bulk-RNA-Sequenzdaten von Tumorproben zu schätzen. Die Methode
beinhaltet das falsche Spezifizieren des Dekonvolutionsmodells, um das
Krebsexpressionssignal in den Modellresiduen anzureichern. In
Simulationsexperimenten konnte ich nachweisen, dass dies für normalverteilte
Daten funktioniert und simuliertes technisches Rauschen toleriert. Auf
simulierten Bulk-RNA-Sequenzdaten erreichte ich immer noch eine hohe
Schätzgenauigkeit, konnte jedoch keine klare Verbesserung gegenüber dem bereits
in den Rohdaten vorhandenen Signal nachweisen. Dasselbe galt für
RNA-Sequenzdaten aus echten Tumorproben. Ebenfalls konnte ich keine Verbesserung
der Vorhersagegenauigkeit feststellen, als ich versucht habe, mithilfe der
geschätzten Expressionsprofile der Krebszellen Vorhersagemodelle für Variablen
auf Tumor- und Patientenebene zu trainieren. Aufgrund ihrer Einfachheit kann die
Methode noch verbessert werden und für die Krebsforschung nützlich werden.

\newpage

\tableofcontents

\newpage

\listoffigures

\newpage

\listoftables

\newpage
